File: C_Programming/Practise_Programs/PP_148.c
//////////////////////////////////////////////////////////////////
//
//  File Name         : PP_148
//
//  Description       : This program uses a bit mask to check whether
//                      the 7th bit of a given unsigned integer is ON
//                      or OFF.
//
//  Author            : Shubham Somanath Gadhe
//  Date              : 18/08/2026
//
//  Time Complexity   : O(1)
//  Space Complexity  : O(1)
//
//////////////////////////////////////////////////////////////////

#include<stdio.h> 

typedef unsigned int UINT;

int main()
{
    UINT iNo = 0;
    UINT iMask = 64;
    UINT iAns = 0;

    printf("Enter Number : \n");
    scanf("%d", &iNo);

    iAns = iNo & iMask;

    if(iAns == iMask)
    {
        printf("7th bit is ON\n");
    }
    else
    {
        printf("7th bit is OFF\n");
    }

    return 0;
}

//////////////////////////////////////////////////////////////////
//
//  Sample Input :
//      64
//
//  Sample Output :
//
//      Enter Number :
//      7th bit is ON
//
//////////////////////////////////////////////////////////////////

File: Java_Programming/Practise_Programs/PP_148.java
//////////////////////////////////////////////////////////////////
//
//  File Name         : PP_148
//
//  Description       : This program prints a centered
//                      increasing star pattern using nested
//                      for loops. The number of spaces decreases
//                      while the number of stars increases in
//                      each successive row.
//
//  Author            : Shubham Somanath Gadhe
//  Date              : 02/09/2026
//
//  Time Complexity   : O(n^2)
//  Space Complexity  : O(1)
//
//////////////////////////////////////////////////////////////////

class PP_148
{
    public static void main(String A[])
    {
        int i, j, k = 0;

        for(i = 1; i <= 5; i++)
        {
            for(j = 1; j <= 5 - i; j++)     // SPACE
            {
                System.out.print(" ");
            }
            for(k = 1; k <= i; k++)     // PRINTING
            {
                System.out.print(" * ");
            }
            System.out.println();
        }
    }
}

//////////////////////////////////////////////////////////////////
//
//  Sample Input :
//      No input required.
//
//  Sample Output :
//
//           *       4
//         * *       3
//       * * *       2
//     * * * *       1
//   * * * * *       0
//
//////////////////////////////////////////////////////////////////